### PR TITLE
refactor(manager): add class "DefaultValue" to "common.py"

### PR DESCRIPTION
**Pull request by graczhual**
To make it easier for the user to understand, use class "DefaultValue" instead
of None to represent the default value of parameters from methods in manager.

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-28v2h00 SDK 版本管理相关函数默认值可读性优化](https://app.clickup.com/t/28v2h00)

File: graviti/manager/common.py [hunk 1]
```diff
@@ -6,12 +6,20 @@
...
 
 if TYPE_CHECKING:
     from graviti.manager.commit import Commit
 
 
+class DatasetStatus(Enum):
+    """An enumeration type representing the current status of the dataset."""
+
+    CURRENT_COMMIT = object()
+    CURRENT_BRANCH = object()
```
Comment (linjiX) [inline graviti/manager/common.py:20]: normally we use `auto()` to generate enum values automatically.
https://docs.python.org/zh-cn/3/library/enum.html#enum.auto

File: graviti/manager/dataset.py [hunk 2]
```diff
@@ -112,8 +110,6 @@ def __init__(
         self.is_public = is_public
         self.config = config
         self._data: Commit = self.branches.get(default_branch)
-        if commit_id != self._data.commit_id:
-            self._data = self.commits.get(commit_id)
```
Comment (linjiX) [inline graviti/manager/dataset.py:116]: shall we use another PR for removing `Dataset.commit_id`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/common.py [hunk 3]
```diff
@@ -6,12 +6,32 @@
...
 
 import warnings
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any
 
 if TYPE_CHECKING:
     from graviti.manager.commit import Commit
 
 
+class ArgumentsDefaultValue:
```
Comment (linjiX) [inline graviti/manager/common.py:15]: I think `DefaultValue` is ok.

File: graviti/manager/draft.py [hunk 4]
```diff
@@ -250,11 +250,14 @@ def __init__(self, dataset: "Dataset") -> None:
 
     def _generate(
         self,
-        state: Optional[str],
+        state: str,
         branch: Optional[str],
```
Comment (linjiX) [inline graviti/manager/draft.py:254]: do we need to change the type of `branch` to `str`?

Comment (graczhual) [reply graviti/manager/draft.py:254]: but we need to set `branch = None` here

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**